lms/static/scripts/frontend_apps/components/SubmitGradeForm.tsx

```diff
@@ -84,6 +85,9 @@ export default function SubmitGradeForm({
   // Used to handle keyboard input changes for the grade input field.
   const inputRef = useRef<HTMLInputElement | null>(null);
 
+  // Make sure instructors are notified if there's a risk to lose unsaved data
+  useWarnOnPageUnload(grade.data !== inputRef.current?.value);
```

Comment:
Only checking inline for the grade value itself for now.

Once we add support for the grading comment we will have to extend this logic, probably by adding a peece of local derived state.

Reply:
The component is not guaranteed to be re-rendered when the input field value changes. To make this work reliably,  this value needs to be computed from props/state.

Reply:
You are right. I remember thinking it would be enough to capture grade changes, but that actually makes no sense.

The component is probably getting re-rendered as a side effect of something else, making this incidentally work, but yeah, this is not correct.

I will probably take the opportunity to define a slightly more elaborated logic, like:

```tsx
const hasUnsavedChanges = useMemo(() => {
  // grade.data is not equal to input value
}, [grade.data, inputValue]);
```

This will be needed for https://github.com/hypothesis/lms/issues/5673, and make even more sense once the comment is in place and hasUnsavedChanges requires more considerations.